### PR TITLE
Fix: signal/slot macro -> func pointer & network fixes

### PR DESCRIPTION
**Pull request by Ryex**
- convert qt connect calls to use function pointers instead of the signal/slot macros wherever practical (UI classes were mostly left alone, target was tasks and processes)
- give signals an explicit receivers to use the static method over the instance method wherever practical
- ensure networks tasks are using the `errorOccured` signal added in Qt5.15 over the deprecated `error` signal
- ensure all networks tasks have an sslErrors signal connected
- add seemingly missing `MinecraftAccount::authSucceeded` connection for `MSAInteractive` login flow
- set `x-xbl-contract-version` header during xbox auth step (GDLauncher and ATLauncher set this)

### CODE REVIEW

**Comment by Ryex** [comment]
This appears to fix a crash in #904.
I also have tentative reports that some of users previously having account issues who tested the CI build from this PR had them resolved.

File: launcher/LoggedProcess.cpp [hunk 1]
```diff
@@ -44,11 +44,11 @@ LoggedProcess::LoggedProcess(QObject *parent) : QProcess(parent)
...
-    connect(this, SIGNAL(finished(int,QProcess::ExitStatus)), SLOT(on_exit(int,QProcess::ExitStatus)));
-#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
-    connect(this, SIGNAL(errorOccurred(QProcess::ProcessError)), this, SLOT(on_error(QProcess::ProcessError)));
+    connect(this, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished), this, &LoggedProcess::on_exit);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 6, 0) // &QProcess::errorOccurred added in 5.6
+    connect(this, &QProcess::errorOccurred, this, &LoggedProcess::on_error);
 #else
-    connect(this, SIGNAL(error(QProcess::ProcessError)), this, SLOT(on_error(QProcess::ProcessError)));
+    connect(this, QOverload<QProcess::ProcessError>::of(&QProcess::error), this, &LoggedProcess::on_error);
 #endif
```
Comment (flowln) [inline launcher/LoggedProcess.cpp]: you can remove the `#if` statement then, we target 5.12 and above, so you won't be able to use the other one anyway.

File: launcher/java/JavaChecker.cpp [hunk 2]
```diff
@@ -87,15 +87,15 @@ void JavaChecker::performCheck()
...
-    connect(process.get(), SIGNAL(finished(int, QProcess::ExitStatus)), this, SLOT(finished(int, QProcess::ExitStatus)));
-#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
-    connect(process.get(), SIGNAL(errorOccurred(QProcess::ProcessError)), this, SLOT(error(QProcess::ProcessError)));
+    connect(process.get(), QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished), this, &JavaChecker::finished);
+#if QT_VERSION >= QT_VERSION_CHECK(5, 6, 0) // &QProcess::errorOccurred added in 5.6
+    connect(process.get(), &QProcess::errorOccurred, this, &JavaChecker::error);
 #else
-    connect(process.get(), SIGNAL(error(QProcess::ProcessError)), this, SLOT(error(QProcess::ProcessError)));
+    connect(process.get(), &QProcess::error, this, &JavaChecker::error);
 #endif
```
Comment (flowln) [inline launcher/java/JavaChecker.cpp]: same here

**Review by flowln — APPROVED** [review]
Noteworth: The fix in #904 is mostly due to adding the `QNetworkReply::errorOccurred` to `Upload`, which is necessary for task abortion to work correctly.

damn there's a lot of code that should probably use our `net/` stuff but doesn't xd

great work here tho! :D

**Comment by DioEgizio** [comment]
Conflicting with develop

**Comment by Ryex** [comment]
fixed

**Review by DioEgizio — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]
Thanks!

**Pull request merged**